diff --git a/notweb/config.example.php b/notweb/config.example.php
index 25a6ac4..eec9a20 100644
--- a/notweb/config.example.php
+++ b/notweb/config.example.php
@@ -4,7 +4,7 @@
 // 1. copy this ./.config.php
 // 2. edit values as required
 
-$config = [];
+$config = array();
 
 // these values will work without any changes, assuming file permissions are ok.
 
diff --git a/notweb/src/RTCStore/Adapter/CSV.php b/notweb/src/RTCStore/Adapter/CSV.php
index 2d1604c..61079c6 100644
--- a/notweb/src/RTCStore/Adapter/CSV.php
+++ b/notweb/src/RTCStore/Adapter/CSV.php
@@ -53,7 +53,7 @@ class CSV implements AdapterInterface
         if (!$fh){
             throw new \Exception("Could not open rtcstore csv file.");
         }
-        $connections = [];
+        $connections = array();
         while ($fh && (FALSE !== ($row = fgetcsv($fh)))){
             if (count($row) == 4){
                 $connection = new Connection();
@@ -87,12 +87,12 @@ class CSV implements AdapterInterface
             throw new \Exception("Could not open rtcstore csv file.");
         }
         foreach ($connections as $connection) {
-            $row = [
+            $row = array(
                 $connection->getName(),
                 $connection->getOffer(),
                 $connection->getAnswer(),
                 serialize($connection->getCandidates())
-            ];
+            );
             fputcsv($fh, $row);
         }
         fclose($fh);
diff --git a/notweb/src/RTCStore/Connection.php b/notweb/src/RTCStore/Connection.php
index c5b2149..cc00fd0 100644
--- a/notweb/src/RTCStore/Connection.php
+++ b/notweb/src/RTCStore/Connection.php
@@ -5,8 +5,15 @@
  * Date: 1/2/16
  * Time: 7:27 PM
  */
+namespace {
+   if (!interface_exists('JsonSerializable')) {
+       interface JsonSerializable
+       {
+       }
+   }
+}
 
-namespace DataFestivus\RTCStore;
+namespace DataFestivus\RTCStore {
 
 /**
  * Class Connection - Models the fields required for an RTC offer
@@ -17,7 +24,7 @@ class Connection implements \JsonSerializable
     private $name = '';
     private $offer = null;
     private $answer = null;
-    private $candidates = [];
+    private $candidates = array();
 
     /**
      * @return string
@@ -82,7 +89,7 @@ class Connection implements \JsonSerializable
     {
         $this->candidates = $answerCandidates;
     }
-    
+
     public function addCandidate($candidate)
     {
         $this->candidates[] = $candidate;
@@ -99,16 +106,18 @@ class Connection implements \JsonSerializable
     {
         $offer = $this->getOffer();
         $answer = $this->getAnswer();
-        $candidates = [];
-        foreach ($this->getCandidates() as $candidate){
+        $candidates = array();
+        foreach ($this->getCandidates() as $candidate) {
             $candidates[] = json_decode($candidate, true);
         }
-        $result = [
+        $result = array(
             'name' => $this->getName(),
             'offer' => ($offer ? json_decode($offer, true) : ''),
             'answer' => ($answer ? json_decode($answer, true) : ''),
             'candidates' => $candidates
-        ];
+        );
         return $result;
     }
+}
+
 }
\ No newline at end of file
diff --git a/web/signal.php b/web/signal.php
index fad0227..1e5d418 100644
--- a/web/signal.php
+++ b/web/signal.php
@@ -21,7 +21,7 @@ if (isset($_SERVER['df_autoload'])){
 }
 include_once(__DIR__ . '/' . $df_autoload);
 
-$CALL_LIST = [CALL_START, CALL_FETCH, CALL_UPDATE];
+$CALL_LIST = array(CALL_START, CALL_FETCH, CALL_UPDATE);
 $name = '';
 $call = '';
 if (array_key_exists('name', $_POST)){
@@ -30,7 +30,7 @@ if (array_key_exists('name', $_POST)){
 if (array_key_exists('call', $_POST)){
     $call = $_POST['call'];
 }
-$errors = [];
+$errors = array();
 if (!$name) {
     $errors['name'] = "'name' parameter is required.";
 }
@@ -41,6 +41,12 @@ $status = 'success';
 $statusCode = 200;
 $rtcConnection = null;
 
+if(!function_exists('\http_response_code')){
+    function http_response_code($code){
+        header("HTTP/1.0 " . $code);
+    }
+}
+
 function df_return()
 {
     global $status, $statusCode, $errors, $rtcConnection;
@@ -50,7 +56,7 @@ function df_return()
     die(json_encode(array(
         'result' => $status,
         'errors' => $errors,
-        'connection' => $rtcConnection
+        'connection' => $rtcConnection ? $rtcConnection->jsonSerialize() : null
     )));
 }
 
@@ -62,13 +68,20 @@ set_exception_handler(function(Exception $ex){
 });
 register_shutdown_function('df_return');
 
+function get_post($key){
+    if (ini_get('magic_quotes_gpc')){
+        return stripslashes($_POST[$key]);
+    } else {
+        return $_POST['key'];
+    }
+}
 
 $store = \DataFestivus\get_store();
 switch ($call) {
     case CALL_START:
         $offer = null;
         if (array_key_exists('connection', $_POST)) {
-            $offer = json_decode($_POST['connection']);
+            $offer = json_decode(get_post('connection'));
         }
         if (!$offer) {
             $errors['connection'] =
@@ -97,7 +110,7 @@ switch ($call) {
         $update = null;
         $rtcConnection = $store->getOffer($name);
         if (array_key_exists('connection', $_POST)) {
-            $update = json_decode($_POST['connection'], true);
+            $update = json_decode(get_post('connection'), true);
         }
         if (!$update){
             $errors['connection'] =
